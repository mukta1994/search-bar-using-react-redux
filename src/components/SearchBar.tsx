import React from 'react';

//material ui
import ClearIcon from '@material-ui/icons/Clear';
import SearchIcon from '@material-ui/icons/Search';


const SearchBar = (props:any) => {

    const {callback, clearSearchInput, onInput, val, onKeyPress}= props

    return(<div className="search-bar">
        <h3>Search effective charitable organizations</h3>
              <div className="search_container">
                <input type="text" className="search-input" onInput={onInput} value={val} onKeyPress={onKeyPress}/>
                    <div className="actions">
                    <ClearIcon className="close-icon" onClick={clearSearchInput}/>
                    <div className="divider"></div>
                    <SearchIcon className="search-icon" onClick={callback}/> 
                </div>  
            </div>
        </div>)
}

export default SearchBar;