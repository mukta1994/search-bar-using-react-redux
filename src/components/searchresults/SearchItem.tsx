import React from 'react';

//material ui
import Card from '@material-ui/core/Card';

//components under search item
import Linearbar from './resultcomponents/Linearbar';
import Images from './resultcomponents/Images';
import OrganizationInfo from './resultcomponents/OrganizationInfo';
import NotMember from './resultcomponents/NotMember';

//types
import { CauseDataType } from '../../types'


export interface OrganizationType {
    organizationData : CauseDataType
}

//Organization card 
const SearchItem:React.FC<OrganizationType> = (props: OrganizationType) => {

    const { organizationData }= props

    return ( <aÂ target="/blank" href={organizationData.website}>
            <Card  className="organization">
      
            {(organizationData.publishedAt !== null && organizationData.hasPassedPreliminary &&
                 <Images image={ organizationData.images.data[0].files.data[0]} logoimage={ organizationData.logo.data.files.data[0] }/>
            )}

             <OrganizationInfo data={organizationData}/>

             {((!organizationData.hasPassedPreliminary || organizationData.publishedAt === null) &&
                 <NotMember/>
             )}

             <Linearbar stage={!organizationData.hasPassedPreliminary ? 33.33 : (organizationData.publishedAt !== null ? 100 : 66)} />

    </Card>
    </a>
    )
}
  

  export default SearchItem;