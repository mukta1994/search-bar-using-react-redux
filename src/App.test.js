import React from 'react';
import { render, fireEvent } from '@testing-library/react';


import SearchBar from './components/SearchBar';
import SearchItem from './components/searchresults/SearchItem';

it("renders correctly", ()=>{
  const { queryByTestId, queryByPlaceholderText } = render (<SearchBar/>)

  expect(queryByTestId("search-icon")).toBeTruthy();

  expect(queryByPlaceholderText("Search")).toBeTruthy();

})

describe("Input value", ()=>{
  it("changes input", ()=>{

    const {queryByPlaceholderText}= render(<SearchBar/>)

    const searchInput = queryByPlaceholderText('Search')
  
    fireEvent.change(searchInput, {target: {value: "test"}})
  
    expect(searchInput.value).toBe("test")

  })
 
})

it('render correctly search item component', () => {  
  const SearchItemComponent = (<SearchItem />);
  expect(SearchItemComponent).toMatchSnapshot();
});










