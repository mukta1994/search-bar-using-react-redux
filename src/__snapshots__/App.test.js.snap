// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly search item component 1`] = `<SearchItem />`;
